wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -991,12 +992,19 @@ impl super::Instance {
         };
 
         if let Some(driver) = phd_capabilities.driver {
+            #[allow(clippy::unnecessary_cast, trivial_numeric_casts)]
+            // False positive on the `as u8` cast on Linux/Android aarch64
```

Comment:
```suggestion
            #[cfg(not(any(target_os = "macos", target_os = "ios")))]
```

I was thinking of silently ignoring the check but if the message has value, we can go with that.

Reply:
Just so that our adapter choosing behavior is clear from the logs.